stats/csv/collector_test.go

```diff
@@ -0,0 +1,328 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package csv
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/loadimpact/k6/stats"
+
+	"github.com/loadimpact/k6/lib"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMakeHeader(t *testing.T) {
+	testdata := map[string][]string{
+		"One tag": {
+			"tag1",
+		},
+		"Two tags": {
+			"tag1", "tag2",
+		},
+	}
+
+	for testname, tags := range testdata {
+		func(testname string, tags []string) {
+			t.Run(testname, func(t *testing.T) {
+				header := MakeHeader(tags)
+				assert.Equal(t, len(tags)+4, len(header))
+				assert.Equal(t, "metric_name", header[0])
+				assert.Equal(t, "timestamp", header[1])
+				assert.Equal(t, "metric_value", header[2])
+				assert.Equal(t, "extra_tags", header[len(header)-1])
+			})
+		}(testname, tags)
+	}
+}
+
+func TestSampleToRow(t *testing.T) {
+	testSamples := []*stats.Sample{
+		{
+			Time:   time.Now(),
+			Metric: stats.New("my_metric", stats.Gauge),
+			Value:  1,
+			Tags: stats.NewSampleTags(map[string]string{
+				"tag1": "val1",
+				"tag2": "val2",
+				"tag3": "val3",
+			}),
+		},
+		{
+			Time:   time.Now(),
+			Metric: stats.New("my_metric", stats.Gauge),
+			Value:  1,
+			Tags: stats.NewSampleTags(map[string]string{
+				"tag1": "val1",
+				"tag2": "val2",
+				"tag3": "val3",
+				"tag4": "val4",
+				"tag5": "val5",
+			}),
+		},
+		nil,
+	}
+
+	enabledTags := map[string][][]string{
+		"One tag": {
+			{"tag1"},
+			{"tag2"},
+		},
+		"Two tags": {
+			{"tag1", "tag2"},
+			{},
+		},
+		"Two tags, one ignored": {
+			{"tag1", "tag2"},
+			{"tag3"},
+		},
+	}
+
+	for testname, tags := range enabledTags {
+		for _, sample := range testSamples {
+			func(testname string, sample *stats.Sample, resTags []string, ignoredTags []string) {
+				t.Run(testname, func(t *testing.T) {
+					row := SampleToRow(sample, resTags, ignoredTags)
+					if row != nil {
+						assert.Equal(t, len(resTags)+4, len(row))
+						for _, tag := range ignoredTags {
+							assert.False(t, strings.Contains(row[len(row)-1], tag))
+						}
+					} else {
+						assert.Nil(t, row)
+						assert.Nil(t, sample)
+					}
+				})
+			}(testname, sample, tags[0], tags[1])
+		}
+	}
+}
+func TestCollect(t *testing.T) {
+	testSamples := []stats.SampleContainer{
+		stats.Sample{
+			Time:   time.Unix(1562324643, 0),
+			Metric: stats.New("my_metric", stats.Gauge),
+			Value:  1,
+			Tags: stats.NewSampleTags(map[string]string{
+				"tag1": "val1",
+				"tag2": "val2",
+				"tag3": "val3",
+			}),
+		},
+		stats.Sample{
+			Time:   time.Unix(1562324644, 0),
+			Metric: stats.New("my_metric", stats.Gauge),
+			Value:  1,
+			Tags: stats.NewSampleTags(map[string]string{
+				"tag1": "val1",
+				"tag2": "val2",
+				"tag3": "val3",
+				"tag4": "val4",
+			}),
+		},
+	}
+	t.Run("Collect", func(t *testing.T) {
```

Comment:
You don't need to call t.Run if you are not going to make at least two subtests in a test. The same goes for the other times in this file where t.Run is just called once in a test. This is not a problem but it's just adding more indentation